taws_minimal/tests/cucumber.rs

```diff
@@ -21,280 +19,204 @@ fn main() {
-        }
-    }
+#[when(expr = "the rate of descent is {constraint} feet per minute")]
+fn when_rate_of_descent(world: &mut MyWorld, rate_of_descent: ConstraintParameter) {
+    let rate_of_descent: Constraint<f64> = rate_of_descent.into();
+
+    let unit = Velocity::new::<velocity::foot_per_minute>(-1.0);
+    let climb_rate = match rate_of_descent {
+        Constraint::AtLeast(p, a) => Constraint::AtMost(p, a * unit),
+        Constraint::AtMost(p, a) => Constraint::AtLeast(p, a * unit),
+        Constraint::Equal(p, a) => Constraint::Equal(p, a * unit),
+        Constraint::InRange(p, a, b) => Constraint::InRange(p, b * unit, a * unit),
+        Constraint::NotInRange(p, a, b) => Constraint::NotInRange(p, b * unit, a * unit),
+    };
+
+    world.constraints[world.phase].add_climb_rate_constraint(climb_rate);
 }
 
-#[when(regex = r"^the height above terrain is ?(not)? between (\d+) and (\d+) feet$")]
-fn height_above_terrain(world: &mut MyWorld, maybe_not: String, lower: f64, upper: f64) {
-    world.update_phase(StepType::When);
-
-    let height_at_least = Length::new::<foot>(lower);
-    let height_at_most = Length::new::<foot>(upper);
-
-    if maybe_not == "not" {
-        world.constraints[world.phase].add_altitude_ground_constraint(Constraint::NotInRange(
-            world.phase,
-            height_at_least,
-            height_at_most,
-        ));
-    } else {
-        world.constraints[world.phase].add_altitude_ground_constraint(Constraint::InRange(
-            world.phase,
-            height_at_least,
-            height_at_most,
-        ));
-        // TODO altitude or altitude_ground
-    }
+#[when(expr = "the height above terrain is {constraint} feet")]
+fn when_height_above_terrain(world: &mut MyWorld, height_above_ground: ConstraintParameter) {
+    let height_above_ground: Constraint<f64> = height_above_ground.into();
+
+    let unit = Length::new::<length::foot>(1.0);
+    let height_above_ground = match height_above_ground {
+        Constraint::AtLeast(p, a) => Constraint::AtLeast(p, a * unit),
+        Constraint::AtMost(p, a) => Constraint::AtMost(p, a * unit),
+        Constraint::Equal(p, a) => Constraint::Equal(p, a * unit),
+        Constraint::InRange(p, a, b) => Constraint::InRange(p, a * unit, b * unit),
+        Constraint::NotInRange(p, a, b) => Constraint::NotInRange(p, a * unit, b * unit),
+    };
+
+    world.constraints[world.phase].add_altitude_ground_constraint(height_above_ground);
 }
 
-#[when(
-    regex = r"^the distance to the nearest airport is ?(not)? between ([0-9]+\.?[0-9]*) and ([0-9]+\.?[0-9]*) NM"
-)]
-fn distance_to_airport(world: &mut MyWorld, maybe_not: String, lower: f64, upper: f64) {
-    world.update_phase(StepType::When);
+#[given(expr = "the plane is {constraint} NM of an airport")]
+#[when(expr = "the distance to the nearest airport is {constraint} NM")]
+fn when_distance_to_airport(_world: &mut MyWorld, _distance: ConstraintParameter) {
     todo!();
```

Comment:
Please don't make a matching function which then contains a `todo!()` macro, because then that step becomes a failing scenario. Rather comment the unfinished step function out, then that scenario becomes a skipped scenario, which is semantically more correct. This of course holds true for all instances of `todo!()` in a step function.